packages/gatsby/src/cache-dir/static-entry.js

```diff
@@ -239,28 +239,32 @@ export default (pagePath, callback) => {
     .slice(0)
     .reverse()
     .forEach(style => {
-      // Add <link>s for styles.
-      headComponents.push(
-        <link
-          as="style"
-          rel={style.rel}
-          key={style.name}
-          href={urlJoin(pathPrefix, style.name)}
-        />
-      )
-
-      headComponents.unshift(
-        <style
-          type="text/css"
-          data-href={urlJoin(pathPrefix, style.name)}
-          dangerouslySetInnerHTML={{
-            __html: fs.readFileSync(
-              join(process.cwd(), `public`, style.name),
-              `utf-8`
-            ),
-          }}
-        />
-      )
+      // Add <link>s for styles that should be prefetched
+      // otherwise, inline as a <style> tag
+
+      if(style.rel === `prefetch`) {
+        headComponents.push(
+          <link
+            as="style"
+            rel={style.rel}
+            key={style.name}
+            href={urlJoin(pathPrefix, style.name)}
+          />
+        )
+      } else {
+        headComponents.unshift(
+          <style
+            type="text/css"
+            data-href={urlJoin(pathPrefix, style.name)}
```

Comment:
Not sure if `data-href` should be removed in this conditional?

Reply:
No that should be there still. Is there so webpack knows the styles have already been loaded so won't try again.